### DISCUSSION TITLE
Unable to start Transmission 4.0.4 after upgrading from 3.0.0

### DISCUSSION THREAD

**Discussion by donacarr** [Freetz-NG/freetz-ng#836] [Allgemein] [answered]
Hi

I built a new freetz image based on 7.56 and main branch for my 7590 box.
My previous build was on 7.29 and transmission 3.0.0
But after the upgrade I got below error:
`Unexpected argument: --no-finish-dir `

![image](https://github.com/Freetz-NG/freetz-ng/assets/13415391/6dce3279-0416-42a1-bc05-cfc522c709b6)

Could you tell me where is the transmission config file?
I tried to delete that parameter from` /var/media/ftp/SandiskDrive/bittorrent2/config/settings.json ` but same error...
I suspect freetz saves that info in some other place...
And do you know how a fresh config file should be?

**Comment by fda77** [comment]
Seems v3 and v4 parameters are changed and are not fixed in freetz yet..

https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/transmission-cgi/files/root/etc/init.d/rc.transmission#L89

You could edit this file on the device temporary by
```
vix /etc/init.d/rc.transmission
```

and then run rc.transmission

**Comment by fda77** [comment, marked as answer]
The "finish-dir" is no upstream option und was not ported to v4, should work now, please test:
* f8e8a5487 transmission-cgi: no finish-dir with v4
* 14b63a76a transmission: addd finish-dir option

If you need this option and you want to maintain it:

 - remove the "depends": https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/transmission/Config.in#L37
 - fix https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/transmission/patches/current/finishdir/120-finish_dir.patch
 - enable in menuconfig
 - run "make transmission-autofix" to check
 - PR

**Comment by donacarr** [comment]
thank you @fda77 
I tried with last branch main and transmission starts correctly.
Then I used the remote GUI to set the finish directory.

**Reply by fda77** [reply]
Thx for feedback. When you need it, send https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/transmission/patches/current/finishdir/120-finish_dir.patch upstream. I dont know who created it, check the git-history of it
